instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SnsCamelTest.groovy

```diff
@@ -185,195 +111,259 @@ class S3CamelTest extends AgentInstrumentationSpecification {
           attributes {
             "aws.agent" "java-aws-sdk"
             "aws.endpoint" String
-            "aws.operation" "ReceiveMessage"
+            "aws.operation" "DeleteMessage"
             "aws.queue.url" queueUrl
             "aws.service" "AmazonSQS"
             "http.flavor" "1.1"
             "http.method" "POST"
             "http.status_code" 200
             "http.url" String
             "net.peer.name" String
-            "net.peer.port" { it == null || Number }
+            "net.peer.port" {it == null || Number}
             "net.transport" "IP.TCP"
           }
         }
       }
-      // camel cleaning received msg
-      trace(5, 1) {
+      // camel polling
+      trace(3, 1) {
         span(0) {
-          name "SQS.DeleteMessage"
+          name "SQS.ReceiveMessage"
           kind CLIENT
           hasNoParent()
           attributes {
             "aws.agent" "java-aws-sdk"
             "aws.endpoint" String
-            "aws.operation" "DeleteMessage"
+            "aws.operation" "ReceiveMessage"
             "aws.queue.url" queueUrl
             "aws.service" "AmazonSQS"
             "http.flavor" "1.1"
             "http.method" "POST"
             "http.status_code" 200
             "http.url" String
             "net.peer.name" String
-            "net.peer.port" { it == null || Number }
+            "net.peer.port" {it == null || Number}
             "net.transport" "IP.TCP"
           }
         }
       }
     }
-
     cleanup:
-    awsConnector.deleteBucket(bucketName)
     awsConnector.purgeQueue(queueUrl)
+    camelApp.stop()
   }
 
-  def waitAndClearSetupTraces(queueUrl, queueName, bucketName) {
-    assertTraces(7) {
+  def waitAndClearSetupTraces(queueUrl, queueName) {
+    assertTraces(6) {
       trace(0, 1) {
+        AwsSpan.sqs(it, 0, "SQS.CreateQueue", queueUrl, queueName)
+      }
+      trace(1, 1) {
+        AwsSpan.sqs(it, 0, "SQS.GetQueueAttributes", queueUrl)
+      }
+      trace(2, 1) {
+        AwsSpan.sqs(it, 0, "SQS.SetQueueAttributes", queueUrl)
+      }
+      trace(3, 1) {
         span(0) {
-          name "SQS.CreateQueue"
+          name "SNS.CreateTopic"
           kind CLIENT
           hasNoParent()
           attributes {
             "aws.agent" "java-aws-sdk"
             "aws.endpoint" String
-            "aws.operation" "CreateQueue"
-            "aws.queue.name" queueName
-            "aws.service" "AmazonSQS"
+            "aws.operation" "CreateTopic"
+            "aws.service" "AmazonSNS"
             "http.flavor" "1.1"
             "http.method" "POST"
             "http.status_code" 200
             "http.url" String
             "net.peer.name" String
-            "net.peer.port" { it == null || Number }
             "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
           }
         }
       }
-      trace(1, 1) {
+      trace(4, 1) {
         span(0) {
-          name "S3.CreateBucket"
+          name "SNS.Subscribe"
```

Comment:
Hmm, maybe for all SNS spans you could add a new `AwsSpan.sns()` method?

Reply:
Yep, that's the idea of a follow-up once I'm sure these changes are heading in the right direction.